premake/wxwidgets.lua

```diff
@@ -525,31 +552,43 @@ function make_filters(libname,libtarget,wxlibs)
     wx_config {Unicode="yes", Version=_OPTIONS["wx_ver"], Static="yes", Debug="yes", WindowsCompiler=wx_compiler, Libs=wxlibs }
   filter { "configurations:Debug", "platforms:Win64" }
     wx_config {Unicode="yes", Version=_OPTIONS["wx_ver"], Static="yes", Debug="yes", Arch="Win64", WindowsCompiler=wx_compiler, Libs=wxlibs }
+  filter { "configurations:Debug", "platforms:MacOsx" }
+    wx_config {Unicode="yes", Version=_OPTIONS["wx_ver"], Static="yes", Debug="yes", Libs=wxlibs, Root="/Users/wangrongwen/cpp/wxtest/wxlib" }
 
   filter { "configurations:Debug wxDLL", "platforms:Win32" }
     wx_config {Unicode="yes", Version=_OPTIONS["wx_ver"], Static="no", Debug="yes", WindowsCompiler=wx_compiler, Libs=wxlibs }
   filter { "configurations:Debug wxDLL", "platforms:Win64" }
     wx_config {Unicode="yes", Version=_OPTIONS["wx_ver"], Static="no", Debug="yes", Arch="Win64", WindowsCompiler=wx_compiler, Libs=wxlibs }
+  filter { "configurations:Debug wxDLL", "platforms:MacOsx" }
+    wx_config {Unicode="yes", Version=_OPTIONS["wx_ver"], Static="no", Debug="yes", Libs=wxlibs, Root="/Users/wangrongwen/cpp/wxtest/wxlib" }
 
   filter { "configurations:Debug DLL", "platforms:Win32" }
     wx_config {Unicode="yes", Version=_OPTIONS["wx_ver"], Static="no", Debug="yes", WindowsCompiler=wx_compiler, Libs=wxlibs }
   filter { "configurations:Debug DLL", "platforms:Win64" }
     wx_config {Unicode="yes", Version=_OPTIONS["wx_ver"], Static="no", Debug="yes", Arch="Win64", WindowsCompiler=wx_compiler, Libs=wxlibs }
+  filter { "configurations:Debug DLL", "platforms:MacOsx" }
+    wx_config {Unicode="yes", Version=_OPTIONS["wx_ver"], Static="no", Debug="yes", Libs=wxlibs, Root="/Users/wangrongwen/cpp/wxtest/wxlib" }
 
   filter { "configurations:Release", "platforms:Win32" }
     wx_config {Unicode="yes", Version=_OPTIONS["wx_ver"], Static="yes", Debug="no", WindowsCompiler=wx_compiler, Libs=wxlibs }
   filter { "configurations:Release", "platforms:Win64" }
     wx_config {Unicode="yes", Version=_OPTIONS["wx_ver"], Static="yes", Debug="no", Arch="Win64", WindowsCompiler=wx_compiler, Libs=wxlibs }
+  filter { "configurations:Release", "platforms:MacOsx" }
+    wx_config {Unicode="yes", Version=_OPTIONS["wx_ver"], Static="yes", Debug="no", Libs=wxlibs, Root="/Users/wangrongwen/cpp/wxtest/wxlib" }
 
   filter { "configurations:Release wxDLL", "platforms:Win32" }
     wx_config {Unicode="yes", Version=_OPTIONS["wx_ver"], Static="no", Debug="no", WindowsCompiler=wx_compiler, Libs=wxlibs }
   filter { "configurations:Release wxDLL", "platforms:Win64" }
     wx_config {Unicode="yes", Version=_OPTIONS["wx_ver"], Static="no", Debug="no", Arch="Win64", WindowsCompiler=wx_compiler, Libs=wxlibs }
+  filter { "configurations:Release wxDLL", "platforms:MacOsx" }
+    wx_config {Unicode="yes", Version=_OPTIONS["wx_ver"], Static="no", Debug="no", Libs=wxlibs, Root="../wxlib" }
```

Comment:
Most likely the relative path for `Root` is also specific for your personal environment setup.